Framework/Core/src/ExternalFairMQDeviceProxy.cxx

```diff
@@ -493,6 +493,12 @@ InjectorFunction dplModelAdaptor(std::vector<OutputSpec> const& filterSpecs, DPL
       }
     }
 
+    int fmqRunNumber = -1;
+    try {
+      fmqRunNumber = atoi(device->fConfig->GetProperty<std::string>("runNumber", "").c_str());
```

Comment:
This will not actually raise any exception as it is undefined behaviour (see https://en.cppreference.com/w/cpp/string/byte/atoi). I would simply use strtoull.

Reply:
The exception would not come from atoi, that will just return 0 in case of a bogus string.
The exception I want to avoid is from fConfig->GetProperty, if that property was not set. In that case we just keep the fmqRunNumber at -1 and ignore it later.